rclcpp/src/rclcpp/node.cpp

```diff
@@ -27,7 +27,17 @@
 #include "rclcpp/node_interfaces/node_clock.hpp"
 #include "rclcpp/node_interfaces/node_graph.hpp"
 #include "rclcpp/node_interfaces/node_logging.hpp"
+// When compiling this file, Windows produces a deprecation warning for the
+// deprecated function prototype of NodeParameters::register_param_change_callback().
```

Comment:
oof, I'll remember this next time I consider deprecating a function.

Can we move these directives around the deprecated function in node_parameters.hpp instead of everywhere it is included?

Reply:
I tried that first, but luckily I didn't trust it and checked, when you do that it prevents the deprecation warning in valid cases too...